src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs

```diff
@@ -39,7 +39,9 @@ internal CSharpRenderingConventions RenderingConventions
 
         internal IList<RazorDiagnostic> Diagnostics { get; } = new List<RazorDiagnostic>();
 
-        internal RazorSourceDocument SourceDocument { get; set; }
+        internal RazorCodeDocument CodeDocument { get; set; }
+
+        internal RazorSourceDocument SourceDocument { get => CodeDocument.Source; }
```

Comment:
`internal RazorSourceDocument SourceDocument { get => CodeDocument.Source; }` can be written as `internal RazorSourceDocument SourceDocument => CodeDocument.Source;`